src/components/Post.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Post as PostUser } from '../types/Post';
+
+type Props = {
+  post: PostUser;
+  setCurrentPost: (post: PostUser | null) => void;
+  currentPost: PostUser | null;
+};
+export const Post: React.FC<Props> = props => {
+  const { post, setCurrentPost, currentPost } = props;
+  const handleOpenDetails = () => {
+    if (!currentPost || currentPost.id !== post.id) {
+      setCurrentPost(post);
+    } else {
+      setCurrentPost(null);
+    }
+  };
+
+  const openClassButton = currentPost?.id === post.id ? '' : 'is-light';
```

Comment:
Don't store classes in a variable, just use [classnames](https://www.npmjs.com/package/classnames) or [clsx](https://www.npmjs.com/package/clsx) library